watcher/lib/influxdb/influxdb.go

```diff
@@ -69,38 +83,100 @@ func (c *Client) Setup() error {
 	for _, query := range queries {
 		log.Infof("%v", query)
 
-		response, err := c.PostForm(c.Endpoint("query"), url.Values{"q": []string{query}})
+		err := c.execQuery(query)
 		if err != nil {
+			if trace.IsAlreadyExists(ConvertInfluxDBError(err)) {
+				log.Info("Retention policy already exists with different attributes.")
+				continue
+			}
 			return trace.Wrap(err)
 		}
-
-		log.Infof("%v %v %v", response.Code(), response.Headers(), string(response.Bytes()))
 	}
 	return nil
 }
 
-// CreateRollup creates a new rollup query in the database
+// CreateRollup creates a rollup query in the database
 func (c *Client) CreateRollup(r Rollup) error {
 	err := r.Check()
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	query, err := buildQuery(r)
+	query, err := r.buildCreateQuery()
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	log.Infof("%v", query)
 
-	response, err := c.PostForm(c.Endpoint("query"), url.Values{"q": []string{query}})
+	if err = c.execQuery(query); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// DeleteRollup deletes a rollup query from the database
+func (c *Client) DeleteRollup(r Rollup) error {
+	err := r.Check()
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	log.Infof("%v %v %v", response.Code(), response.Headers(), string(response.Bytes()))
+	query, err := r.buildDeleteQuery()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	log.Infof("%v", query)
+
+	if err = c.execQuery(query); err != nil {
+		return trace.Wrap(err)
+	}
 	return nil
 }
 
+// UpdateRollup updates a rollup query in the database
```

Comment:
What if the rollup query does not exist - will it fail? It would be helpful to know which preconditions there're.